sdk/core/src/context.rs

```diff
@@ -1,11 +1,177 @@
+use std::any::{Any, TypeId};
+use std::collections::HashMap;
+use std::sync::Arc;
+
 /// Pipeline execution context.
-#[derive(Clone, Debug, Default)]
+#[derive(Clone, Debug)]
 pub struct Context {
-    _priv: (),
+    pub type_map: HashMap<TypeId, Arc<dyn Any + Send + Sync>>,
```

Comment:
Can we keep this private at least for now? This makes it much easier to change implementation in the future. This also opens up for the user accidentally corrupting the map like `ctx.type_map.insert(TypeId::of::<u32>, "hello, world");` which could cause panics or other bugs.

Reply:
Ah yes you are right! Fixing right away! 👍